static/js/compose_actions.js

```diff
@@ -341,7 +341,12 @@ exports.on_topic_narrow = function () {
         // appropriate (after all, they were starting to
         // compose on the old topic and may now be looking
         // for info), so we punt and cancel.
-        exports.cancel();
+
+        // If subject is not same as topic narrowed to then
+        // stop composing
+        if (compose_state.subject() !== narrow_state.topic()) {
```

Comment:
We should compare the topics in lower-case form (since they are case-insensitive in Zulip).  Otherwise lgtm.